src/app/zap-templates/zcl/data-model/chip/global-structs.xml

```diff
@@ -25,6 +25,7 @@ TODO: Make these structures global rather than defining them for each cluster.
   <domain name="CHIP"/>
   <struct name="HomeLocationStruct">
```

Comment:
We really need to fix this to actually follow the spec and be a standalone global type, not tied to a bunch of specific clusters.... I feel like people committed to doing this before introducing it, but no one is actually doing it.

Reply:
What needs to be done is:

1. We need a way of notating global enums and structs in the XML.  One plausible way is "struct or enum definition not associated with any cluster ids".
2. We need to decide what we want the generated code for these things to look like (e.g. which namespace they should live in).
3. We need to modify the cluster-objects codegen templates to do that. Right now, it has bits for "enum/struct associated with more than one cluster" and "enum/struct associated with exactly one cluster"; we would have a bit for "global enum/struct" (detected as "associated with no clusters").

It's not a huge amount of work; probably less work than the amount of workaround we have done now.

Reply:
Er, actually, I forgot two things:

4. The codegen in various places will need to be modified to handle global things correctly instead of assuming that all type names are imported into the namespace of every cluster that uses those types.  This part will require some work. 
5. The various API codegen bits (Darwin, Python, Android) will need to be fixed to expose the global bits correctly.  I'm happy to do the Darwin part, obviously.

Reply:
I don't think I will be able to take this on before the July 29th deadline

Reply:
Yes, it's a lot of work.  People promised me to do that work back in May, when they initially added global-structs.xml....

This PR is fine as far as it goes, but we really need to fix this global stuff before ship.